checkstyle/checkstyle.xml

```diff
@@ -77,6 +77,12 @@
       <property name="validateEnhancedForLoopVariable" value="true"/>
     </module>
 
+    <!-- if/else/for/while/do blocks should use braces (suppressed except for Streams) -->
+    <module name="NeedBraces">
+      <property name="tokens" value="LITERAL_IF, LITERAL_ELSE, LITERAL_FOR, LITERAL_WHILE, LITERAL_DO"/>
```

Comment:
It's a real shame that we have this inconsistency. And it only makes the code worse, this PR being a good example.

Reply:
I am happy to enable this check style for the whole code base (similar to `final` requirement). But not sure if we get some push back -- thus, it seems to be better to introduce those things step by step?

Reply:
I was suggesting the opposite. What's the justification for this rule? At least for `final`, there is the mutability argument. The truth is that Java is becoming a more modern language and redundant verbosity is being eliminated (an extreme example that has been proposed http://openjdk.java.net/jeps/8209434). Requiring braces in every situation perhaps made sense for the old Java. :)

Reply:
For me personally, it's readability question -- I think, it improves readability.

Also, this rule only required braces for if/else and loop-blocks. Not for _every_ situation.

Reply:
Hmm, what other situation is there where braces are optional?

Reply:
switch-case statement

Reply:
http://checkstyle.sourceforge.net/config_blocks.html#NeedBraces